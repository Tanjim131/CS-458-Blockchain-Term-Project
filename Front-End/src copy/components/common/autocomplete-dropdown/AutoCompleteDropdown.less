//less for form auto complete dropdown field
.auto-complete-dropdwon {
  .ant-select-single:not(.ant-select-customize-input) .ant-select-selector, .ant-input-affix-wrapper, .ant-picker {
    height: 40px;
  }
  .ant-picker {
    width: 100%;
  }
  .ant-select-single .ant-select-selector .ant-select-selection-item,
  .ant-select-single .ant-select-selector .ant-select-selection-placeholder {
    line-height: 40px;
  }
  .ant-form-item-label > label {
    color: #788CA0;
  }
  .ant-form-item-has-error .ant-select:not(.ant-select-disabled):not(.ant-select-customize-input) .ant-select-selector {
    border-color: #D24B19 !important;
  }
  .ant-form-item-explain.ant-form-item-explain-error {
    color: #D24B19;
  }
}