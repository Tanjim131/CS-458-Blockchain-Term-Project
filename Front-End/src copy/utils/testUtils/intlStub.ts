import { createIntl } from "react-intl";
const en_us = require("../../i18n/translations/en.json");

export const intlStub = createIntl({ locale: "en-us", messages: en_us });
export const intlMessages = en_us;
