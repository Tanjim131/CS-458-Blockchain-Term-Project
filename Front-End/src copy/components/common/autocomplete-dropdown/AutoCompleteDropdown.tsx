import React, { FC } from "react";
import { Form, Select } from "antd";
import { InfoCircleOutlined } from "@ant-design/icons";
import { injectIntl, IntlShape } from "react-intl";

import "./AutoCompleteDropdown.less";

/**
 * Interface for programme details.
 */
interface IAutoCompleteDropdown {
  intl: IntlShape;
  formItemName: string;
  formItemLabel: string;
  rulesMsg: string;
  required: boolean;
  tooltipTitle?: string;
  placeholderText: string;
  onChangeEvent?: (val: string) => void;
  clearField: boolean;
  optionList: any;
}

/**
 * This is used for AutoCompleteDropdown
 *
 * @usage: <AutoCompleteDropdown formItemName={string} formItemLabel={string} rulesMsg={string} required={boolean} tooltipTitle={string} placeholderText={string} onChangeEvent={function} clearField={boolean} optionList={array} />
 */
const AutoCompleteDropdown: FC<IAutoCompleteDropdown> = (props: IAutoCompleteDropdown) => {
  const { intl, formItemName, formItemLabel, rulesMsg = intl.formatMessage({ id: "upload_file_error" }), required = true, tooltipTitle, placeholderText, onChangeEvent, clearField = false, optionList = [] } = props;

  return (
    <>
      <Form.Item
        name={formItemName}
        label={formItemLabel}
        className="auto-complete-dropdwon"
        rules={[
          {
            required,
            message: rulesMsg,
          },
        ]}
        tooltip={tooltipTitle ? {
          title: tooltipTitle,
          icon: <InfoCircleOutlined />,
          placement: "rightTop",
          autoAdjustOverflow: true,
          overlayClassName: "tooltip-add-document",
        } : ""}
      >
        <Select
          showSearch={true}
          placeholder={placeholderText}
          optionFilterProp="children"
          allowClear={clearField}
          onChange={onChangeEvent}
          filterOption={(input, option) =>
            option?.children.toLowerCase().indexOf(input.toLowerCase()) >= 0
          }
        >
          {(optionList && optionList.length > 0 ) &&
            optionList?.map((documt: any) => (
              <Select.Option key={documt.id} value={documt.id}>
                {documt.name}
              </Select.Option>
            ))}
        </Select>
      </Form.Item>
    </>
  );
};

export default injectIntl(AutoCompleteDropdown);
