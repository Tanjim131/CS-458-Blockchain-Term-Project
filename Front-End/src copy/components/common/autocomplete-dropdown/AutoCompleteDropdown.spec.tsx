import React from 'react';
import { shallow } from "enzyme";
import AutoCompleteDropdown from "./AutoCompleteDropdown";
import { shallowToJson } from "enzyme-to-json";
import { intlStub } from '../../../../utils/testUtils/intlStub';

describe('AutoCompleteDropdown Component', () => {
    let wrapper: any;
    const mockFunc = jest.fn();

    beforeAll(() => {
        wrapper = shallow(
            <AutoCompleteDropdown.WrappedComponent
            intl = {intlStub}
            formItemName = "name"
            formItemLabel = "label"
            rulesMsg = "message"
            required = {true}
            tooltipTitle = "anytitle"
            placeholderText = "any text"
            onChangeEvent = {mockFunc}
            clearField = {true}
            optionList = {[]}
            />
        )
    })

    it('Check if component exists or not and capture snapshot', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    })

})