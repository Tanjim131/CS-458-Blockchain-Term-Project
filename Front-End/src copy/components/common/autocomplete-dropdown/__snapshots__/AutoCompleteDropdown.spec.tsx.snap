// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutoCompleteDropdown Component Check if component exists or not and capture snapshot 1`] = `
<Fragment>
  <FormItem
    className="auto-complete-dropdwon"
    label="label"
    name="name"
    rules={
      Array [
        Object {
          "message": "message",
          "required": true,
        },
      ]
    }
    tooltip={
      Object {
        "autoAdjustOverflow": true,
        "icon": <ForwardRef(InfoCircleOutlined) />,
        "overlayClassName": "tooltip-add-document",
        "placement": "rightTop",
        "title": "anytitle",
      }
    }
  >
    <ForwardRef(InternalSelect)
      allowClear={true}
      filterOption={[Function]}
      onChange={[MockFunction]}
      optionFilterProp="children"
      placeholder="any text"
      showSearch={true}
    />
  </FormItem>
</Fragment>
`;
